src/cfml/system/endpoints/Maven.cfc

```diff
@@ -0,0 +1,701 @@
+		var addr       = getGlobalRepos().mavenCentral & replace( groupId, ".", "/", "ALL" ) & "/" & artifactId & "/" & version & "/" & artifactId & "-" & version & ".pom";
+		var httpResult = "";
+
+		if ( configService.getSetting( "offlineMode", false ) ) {
+			throw(
+				"Can't download [#getNamePrefixes()#:#artifactId#], CommandBox is in offline mode.  Go online with [config set offlineMode=false].",
+				"endpointException"
+			);
+		}
+
+		cfhttp(
+			url         = "#addr#",
+			proxyServer = "#configService.getSetting( "proxy.server", "" )#",
+			method      = "get",
+			redirect    = true,
+			result      = "httpResult"
+		);
+		if ( httpResult.statusCode contains "200" ) {
+			return parsePOM( httpResult.fileContent );
+		} else {
+			throw( message = "Repository Request to #addr# returned status: #httpResult.statusCode#" );
+		}
+	}
+
+	/**
+	 * Get the URLs for the artifact and its dependencies
+	 * @groupId The group ID of the artifact
+	 * @artifactId The artifact ID
+	 * @version The version of the artifact
+	 * @scopes The scopes to include
+	 * @depth The depth of the dependencies
+	 */
+	private function getArtifactAndDependencyJarURLs(
+		groupId,
+		artifactId,
+		version,
+		scopes = "runtime,compile",
+		depth  = 0
+	){
+		var meta   = getArtifactVersion( groupId, artifactId, version );
+		var cache  = {};
+		var result = [];
+		var dep    = "";
+		var d      = "";
+		var v      = "";
+		if ( meta.packaging IS "jar" ) {
+			result = [
+				{
+					"download"   : getJarFileURL( groupId, artifactId, version ),
+					"groupId"    : arguments.groupId,
+					"artifactId" : arguments.artifactId,
+					"version"    : arguments.version
+				}
+			];
+		}
+		for ( dep in meta.dependencies ) {
+			if ( !listFindNoCase( arguments.scopes, dep.scope ) ) {
+				// skip
+				continue;
+			}
+			if ( dep.optional ) {
+				continue;
+			}
+			if ( !cache.keyExists( dep.groupId & "/" & dep.artifactId ) ) {
+				d = getArtifactMetadataFromMaven( dep.groupId, dep.artifactId );
+				if ( len( dep.version ) ) {
+					d.wantedVersion = [ dep.version ];
+				}
+				cache[ dep.groupId & "/" & dep.artifactId ] = d;
+			} else if ( len( dep.version ) ) {
+				// add as a wanted version
+				arrayAppend( cache[ dep.groupId & "/" & dep.artifactId ].wantedVersion, dep.version );
+			}
+		}
+
+		for ( dep in cache ) {
+			dep = cache[ dep ];
+			if ( !dep.keyExists( "wantedVersion" ) ) {
+				v = dep.versioning.release;
+			} else {
+				// todo pick highest version
+				v = dep.wantedVersion[ 1 ];
+			}
+			if ( dep.artifactId == arguments.artifactId && dep.groupId == arguments.groupId ) {
+				continue;
+			}
+			if ( meta.packaging IS "pom" && dep.scope IS "import" ) {
+				if ( depth > 10 ) {
+					throw( message = "Maximum depth of 10 reached" );
+				}
+				d = getArtifactAndDependencyJarURLs(
+					dep.groupId,
+					dep.artifactId,
+					v,
+					scopes,
+					depth++
+				);
+				for ( v in d ) {
```

Comment:
Please use full words for variable names.

Reply:
done